clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -133,6 +144,25 @@ public Struct toStruct() {
     }
 
     @Override
+    public String toString() {
```

Comment:
If we match the old ProducerRecord behavior and don't include the actual record data in the toString (which I think we should change), then your proposal ends up essentially matching what I said.